components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/IdentityCoreServiceComponent.java

```diff
@@ -147,6 +147,19 @@ protected void activate(ComponentContext ctxt) {
                 log.error("Error occurred while initializing the migration client." , e);
             }
 
+            try {
+                Class<?> migrationClientClass = Class.forName(MIGRATION_SERVICE_CLIENT_CLASS_NAME);
+                Object migrationClientObject = migrationClientClass.newInstance();
+                if (migrationClientObject != null) {
+                    log.info("Migration Service class is loaded successfully and going start the process.");
+                }
+                migrationClientClass.getMethod("execute").invoke(migrationClientObject);
+            } catch (Throwable e) {
+                if (log.isDebugEnabled()) {
```

Comment:
This is an exceptional case, you have to provide a reason about why you are doing this.

Reply:
These are changed in new solution.